### PR TITLE
Add a fail point that can hold PS snapshot for five minute.

### PR DESCRIPTION
**Pull request by jiaqizho**
### What problem does this PR solve?

Issue Number: ref #3594 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp [hunk 1]
```diff
@@ -1031,9 +1032,17 @@ BlockInputStreams DeltaMergeStore::readRaw(const Context & db_context,
...
                 tasks.push_back(std::make_shared<SegmentReadTask>(segment, segment_snap, RowKeyRanges{segment->getRowKeyRange()}));
             }
         }
+
+        fiu_do_on(FailPoints::force_slow_page_storage_snapshot_release, {
+            std::thread thread_hold_snapshots([tasks]() {
+                std::this_thread::sleep_for(std::chrono::seconds(5 * 60));
+                (void)tasks;
+            });
+        });
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1045]: We should do this outside the shared_lock on `read_write_mutex`

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp [hunk 2]
```diff
@@ -1031,11 +1032,19 @@ BlockInputStreams DeltaMergeStore::readRaw(const Context & db_context,
...
                 tasks.push_back(std::make_shared<SegmentReadTask>(segment, segment_snap, RowKeyRanges{segment->getRowKeyRange()}));
             }
         }
     }
 
+    fiu_do_on(FailPoints::force_slow_page_storage_snapshot_release, {
+        std::thread thread_hold_snapshots([tasks]() {
+            std::this_thread::sleep_for(std::chrono::seconds(5 * 60));
+            (void)tasks;
+        });
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1045]: This creates a std::thread object without calling `deatch`/`join`, will it make TiFlash crash if we enable this failpoint?
https://en.cppreference.com/w/cpp/thread/thread/~thread

Should we call `deatch` on the thread object?

Comment (jiaqizho) [reply dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1050]: oh, it is diff with `pthread`. added.

**Review by JaySon-Huang — APPROVED** [review]
lgtm

**Comment by jiaqizho** [comment]
@lidezhu PTAL

**Review by lidezhu — APPROVED** [review]
LGTM

**Pull request merged**